src/VisualStudio/CSharp/Impl/Options/Formatting/CodeStylePage.cs

```diff
@@ -11,7 +18,406 @@ internal class CodeStylePage : AbstractOptionPage
+            // csharp_preserve_single_line_statements
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.WrappingKeepStatementsOnSingleLine, editorconfig);
+            // csharp_preserve_single_line_blocks
+            CSharpFormattingOptions_GenerateEditorconfig(optionSet, CSharpFormattingOptions.WrappingPreserveSingleLine, editorconfig);
+        }
+
+        private static void CSharpFormattingOptions_GenerateEditorconfig(OptionSet optionSet, Option<bool> option, StringBuilder editorconfig)
+        {
+            var element = (EditorConfigStorageLocation<bool>)option.StorageLocations.OfType<IEditorConfigStorageLocation>().SingleOrDefault();
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+
+                editorconfig.AppendLine(optionSet.GetOption(CSharpFormattingOptions.NewLineForElse).ToString().ToLower());
+            }
+        }
+
+        private static void CSharpCodeStyleOptions_GenerateEditorconfig(OptionSet optionSet, PerLanguageOption<CodeStyleOption<bool>> option, StringBuilder editorconfig)
+        {
+            var element = (EditorConfigStorageLocation<CodeStyleOption<bool>>)option.StorageLocations.OfType<IEditorConfigStorageLocation>().SingleOrDefault();
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+
+                var curSetting = optionSet.GetOption(option, LanguageNames.CSharp);
+                editorconfig.AppendLine(curSetting.Value.ToString().ToLower() + ":" + curSetting.Notification.ToString().ToLower());
+            }
+        }
+
+        private static void CSharpCodeStyleOptions_GenerateEditorconfig(OptionSet optionSet, Option<CodeStyleOption<bool>> option, StringBuilder editorconfig)
+        {
+            var element = (EditorConfigStorageLocation<CodeStyleOption<bool>>)option.StorageLocations.OfType<IEditorConfigStorageLocation>().SingleOrDefault();
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+
+                var curSetting = optionSet.GetOption(option);
+                editorconfig.AppendLine(curSetting.Value.ToString().ToLower() + ":" + curSetting.Notification.ToString().ToLower());
+            }
+        }
+
+        private static void CSharpCodeStyleOptions_GenerateEditorconfig(OptionSet optionSet, Option<CodeStyleOption<string>> option, StringBuilder editorconfig)
+        {
+            var element = (EditorConfigStorageLocation<CodeStyleOption<string>>)option.StorageLocations.OfType<IEditorConfigStorageLocation>().SingleOrDefault();
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+
+                var curSetting = optionSet.GetOption(option);
+                editorconfig.AppendLine(curSetting.Value.ToString().ToLower() + ":" + curSetting.Notification.ToString().ToLower());
```

Comment:
note: this method is identical to the above method.  using generics, we could share these entirely.   But it's minor.  If you would like any help on this, let me know.